docs/nddata/nddata.rst

```diff
@@ -6,140 +6,247 @@ NDData
+    Unit("m")
 
-    >>> ndd3 = NDData([1, 2, 3, 4])
+the unit of the `~astropy.units.Quantity` is being ignored and the unit is set
+to the explicitly passed one.
 
-The final way to instantiate an `~astropy.nddata.NDData` object is with a data
-object that presents a numpy array-like interface. If the object passed to the
-intializer has all three of the attributes ``shape``, ``__getitem__`` (so it
-is indexable) and ``__array__`` (so that it can act like a numpy array in
-expression) then the ``data`` attribute will be set to that object.
+It might be possible to pass other classes as ``data`` parameter as long as
+they have the properties ``shape``, ``dtype``, ``__getitem__`` and
+``__array__``.
 
 The purpose of this mechanism is to allow considerable flexibility in the
-objects used to store the data while providing a useful to default (numpy
-array).
+objects used to store the data while providing a useful default (numpy array).
 
 Mask
 ----
 
-Values can be masked using the ``mask`` attribute.  One straightforward way to
-provide a mask is to use a boolean numpy array::
+The ``mask`` is being used to indicate if data points are valid or invalid.
+`~astropy.nddata.NDData` doesn't restrict this mask in any way but it is
+recommended to adapt the `numpy.ma.MaskedArray` convention that the mask:
+
++ returns ``True`` for data points that are considered **invalid**.
++ returns ``False`` for those points that are **valid**.
+
+one possibility is to create a mask by using numpys comparison operators::
+
+    >>> array = np.array([0, 1, 4, 0, 2])
+
+    >>> mask = array == 0  # Mask points containing 0
+    >>> mask
+    array([ True, False, False,  True, False], dtype=bool)
+
+    >>> other_mask = array > 1  # Mask points with a value greater than 1
+    >>> other_mask
+    array([False, False,  True, False,  True], dtype=bool)
 
-     >>> ndd_masked = NDData(ndd, mask = ndd.data > 0.9)
+and initialize the `~astropy.nddata.NDData` instance using the ``mask``
+parameter::
 
-Another is to simply initialize an `~astropy.nddata.NDData` object  with a
-masked numpy array::
+    >>> ndd = NDData(array, mask=mask)
+    >>> ndd.mask
+    array([ True, False, False,  True, False], dtype=bool)
 
-    >>> masked_array = np.ma.array([1, 2, 3, 4], mask=[1, 0, 0, 1])
-    >>> ndd_masked = NDData(masked_array)
-    >>> ndd_masked.mask
-    array([ True, False, False,  True], dtype=bool)
+or by replacing the mask::
 
-A mask value of `True` indicates a value that should be ignored, while a mask
-value of `False` indicates a valid value.
+    >>> ndd.mask = other_mask
+    >>> ndd.mask
+    array([False, False,  True, False,  True], dtype=bool)
 
-There is no requirement that the mask actually be a numpy array; for example,
-a function which evaluates a mask value as needed is acceptable as long as it
-follows the convention that `True` indicates a value that should be ignored.
+but that it is a `numpy.ndarray` is not a requirement but probably the most
```

Comment:
Do you mean the whole sentence: "There is no requirement that the mask actually be a numpy array; for example, a function which evaluates a mask value as needed is acceptable as long as it follows the convention that `True` indicates a value that should be ignored."?